File: src/02.string/04.transform/02.412.go
package transform

import "strconv"

func FizzBuzz(n int) []string {

	res := []string{}

	for i := 1; i <= n; i++ {
		if i%15 == 0 {
			res = append(res, "FizzBuzz")
		} else if i%3 == 0 {
			res = append(res, "Fizz")
		} else if i%5 == 0 {
			res = append(res, "Buzz")
		} else {
			res = append(res, strconv.Itoa(i))
		}
	}

	return res
}

File: src/02.string/04.transform/02.412.ts
export function fizzBuzz(n: number): string[] {
    const res = [];
    for (let i = 1; i <= n; i++) {
        if (i % 5 === 0 && i % 3 === 0) {
            res.push("FizzBuzz");
        } else if (i % 3 === 0) {
            res.push("Fizz");
        } else if (i % 5 === 0) {
            res.push("Buzz");
        } else {
            res.push(String(i));
        }
    }

    return res;
}
